### PR TITLE
[zh-cn]: update the translation of statement overview

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

As the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#18266

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 1]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
+  - : 用来引入外部的模块或另一个脚本中导出的函数。
 - {{jsxref("Statements/label", "label")}}
-  - : 带标识的语句，与 `break` 或 `continue` 语句一起使用。
+  - : 提供带标识的语句，与 `break` 或 `continue` 语句一起使用。
 - {{jsxref("Statements/with", "with")}} {{deprecated_inline}}
   - : 拓展一个语句的作用域。
 
+## 语句和声明的区别
+
+在本节内容中，我们将混合两种结构：[_语句_](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#prod-Statement) 和 [_声明_](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#prod-Declaration)。它们是两套不相关的语法。以下是声明：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/index.md:91]:
```suggestion
在本节内容中，我们将混合两种结构：[_语句_](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#prod-Statement)和[_声明_](https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#prod-Declaration)。它们是两套不相关的语法。以下是声明：
```

**Review by yin1999 — COMMENTED** [review]
Could you fix those broken links: https://pr18275.content.dev.mdn.mozit.cloud/zh-CN/docs/Web/JavaScript/Reference/Statements#_flaws

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 2]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
-  - : 在异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)上迭代，调用自定义迭代钩子，其中包含要为每个不同属性的值执行的语句。
+  - : 遍历异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)，对每个不同的属性的值，调用一个自定义的带有要执行的语句的迭代钩子。
 - {{jsxref("Statements/while", "while")}}
-  - : 创建一个循环语句，循环会一直持续到该语句条件表达式的值为 false。先执行条件表达式，然后执行语句。
+  - : 创建一个执行指定语句的循环，直到测试条件的求值为假。先对条件进行判断，再执行语句。
 
 ### 其他
 
+- {{jsxref("Statements/Empty", "Empty", "", 1)}}
+  - ：空语句，用于 JavaScript 期望语句语法的地方表示没有语句。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:73]: format fixing and translation:

```suggestion
  - : 空语句，用于不在 JavaScript 语法期望有语句地方提供语句。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 3]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
 - {{jsxref("Statements/for-await...of", "for await...of")}}
-  - : 在异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)上迭代，调用自定义迭代钩子，其中包含要为每个不同属性的值执行的语句。
+  - : 遍历异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)，对每个不同的属性的值，调用一个自定义的带有要执行的语句的迭代钩子。
 - {{jsxref("Statements/while", "while")}}
-  - : 创建一个循环语句，循环会一直持续到该语句条件表达式的值为 false。先执行条件表达式，然后执行语句。
+  - : 创建一个执行指定语句的循环，直到测试条件的求值为假。先对条件进行判断，再执行语句。
 
 ### 其他
 
+- {{jsxref("Statements/Empty", "Empty", "", 1)}}
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:72]: I think we can translate this part

```suggestion
- {{jsxref("Statements/Empty", "空语句", "", 1)}}
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 4]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
+  - ：空语句，用于 JavaScript 期望语句语法的地方表示没有语句。
+- {{jsxref("Statements/block", "Block", "", 1)}}
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:74]:
```suggestion
- {{jsxref("Statements/block", "块语句", "", 1)}}
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 5]
```diff
@@ -13,29 +15,27 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
 
 ### 控制流程
 
-- {{jsxref("Statements/block", "Block")}}
-  - : 一个块语句可以用来管理零个或多个语句。该区块是由一对大括号分隔。
+- {{jsxref("Statements/return", "return")}}
+  - : 指定函数要返回的值。
 - {{jsxref("Statements/break", "break")}}
-  - : 终止当前的循环，switch 或 label 语句，使程序跳到下一个语句执行。
+  - : 终止当前的循环，switch 或标签语句，使程序跳到下一个语句执行。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:21]:
```suggestion
  - : 终止当前的循环，switch 或标签语句，使程序跳到终止的语句之后的语句执行。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 6]
```diff
@@ -13,29 +15,27 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
-  - : 空语句用来表示没有语句的情况，尽管 JavaScript 语法期望有语句提供。
-- {{jsxref("Statements/if...else", "if...else")}}
-  - : 如果指定的条件是 true，则执行相匹配的一个语句，若为 false，则执行另一个语句。
-- {{jsxref("Statements/switch", "switch")}}
-  - : 计算表达式，将子句于表达式的值做匹配，执行与该值相关联的语句。
+  - : 终止当前或标签循环的当前迭代，直接执行下一个迭代循环。
 - {{jsxref("Statements/throw", "throw")}}
   - : 抛出一个用户定义的异常。
+- {{jsxref("Statements/if...else", "if...else")}}
+  - : 如果指定的条件为真，则执行相匹配的一个语句，若为假，则执行另一个语句。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:27]:
```suggestion
  - : 如果指定的条件为真，则执行相匹配的一个语句；若为假，则执行另一个语句。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 7]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
   - : 无序遍历对象的可枚举属性。语句针对每个唯一的属性。
 - {{jsxref("Statements/for...of", "for...of")}}
-  - : 遍历可迭代的对象（包括 {{jsxref("Global_Objects/Array","数组")}}、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators))，对每个不同属性的属性，调用一个自定义的有执行语句的迭代钩子。
+  - : 遍历可迭代对象（包括 {{jsxref("Global_Objects/Array","数组")}}、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)），对每个不同的属性的值，调用一个自定义的带有要执行的语句的迭代钩子。
 - {{jsxref("Statements/for-await...of", "for await...of")}}
-  - : 在异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)上迭代，调用自定义迭代钩子，其中包含要为每个不同属性的值执行的语句。
+  - : 遍历异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)，对每个不同的属性的值，调用一个自定义的带有要执行的语句的迭代钩子。
 - {{jsxref("Statements/while", "while")}}
-  - : 创建一个循环语句，循环会一直持续到该语句条件表达式的值为 false。先执行条件表达式，然后执行语句。
+  - : 创建一个执行指定语句的循环，直到测试条件的求值为假。先对条件进行判断，再执行语句。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:68]:
```suggestion
  - : 创建一个执行指定语句的循环，直到测试条件的求值为假。先对条件进行求值，再执行语句。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 8]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
 
 ### 其他
 
+- {{jsxref("Statements/Empty", "Empty", "", 1)}}
+  - ：空语句，用于 JavaScript 期望语句语法的地方表示没有语句。
+- {{jsxref("Statements/block", "Block", "", 1)}}
+  - : 块语句，用于将零个或多个语句组合在一起。块由一对花括号括起来。
+- {{jsxref("Statements/Expression_statement", "Expression statement", "", 1)}}
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:76]:
```suggestion
- {{jsxref("Statements/Expression_statement", "表达式语句", "", 1)}}
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 9]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
+  - : 表达式语句，计算表达式的值，但不使用其结果。它允许表达式执行副作用，例如执行函数或更新变量。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:77]:
```suggestion
  - : 表达式语句计算表达式的值并丢弃其结果。它允许表达式产生副作用，例如执行函数或更新变量。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 10]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
+术语“语句”和“声明”在 JavaScript 的正式语法中具有精确的含义，这会影响它们在代码中的位置。例如，在大多数控制流程结构中，其主体只接受语句——如 [`if...else`](/zh-CN/docs/Web/JavaScript/Reference/Statements/if...else) 的两个分支：
+
+```js-nolint
+if (condition)
+  statement1;
+else
+  statement2;
+```
+
+如果使用声明而不是语句，则会产生{{jsxref("SyntaxError", "语法错误")}}。例如，[`let`](<(/zh-CN/docs/Web/JavaScript/Reference/Statements/let)>) 声明不是语句，所以不能将其用作 `if` 语句的主体。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:114]:
```suggestion
如果使用声明而不是语句，则会产生{{jsxref("SyntaxError", "语法错误", "", 1)}}。例如，[`let`](/zh-CN/docs/Web/JavaScript/Reference/Statements/let) 声明不是语句，所以不能将其用作 `if` 语句的主体。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 11]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
+
+你可以将声明看作“{{Glossary("binding", "绑定")}}标识符到值”的过程，而语句则是“执行操作”的过程。`var` 语句的特殊性在于它不遵循正常的词法作用域规则，这可能会产生副作用——例如，创建全局变量、修改已存在的 `var` 定义的变量、定义可在其块之外可见的变量（这是因为 `var` 定义的变量并不是块作用域的）。
+
+在另一个例子中，[标签](/zh-CN/docs/Web/JavaScript/Reference/Statements/label)只能附加到语句上。
+
+```js-nolint example-bad
+label: const a = 1; // SyntaxError: Lexical declaration cannot appear in a single-statement context
+```
+
+> **备注：** 有一种遗留语法允许[函数声明具有标签](/zh-CN/docs/Web/JavaScript/Reference/Statements/label#标记函数声明)，但它只是为了与网络现实兼容而进行了标准化。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:136]:
```suggestion
> **备注：** 有一种遗留语法允许[函数声明具有标签](/zh-CN/docs/Web/JavaScript/Reference/Statements/label#标记函数声明)，但它只是为了与 web 现实兼容而进行了标准化。
```

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 12]
```diff
@@ -44,44 +44,109 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
+}
+
+if (condition) {
+  let i = 0;
+}
+```
+
 ## 浏览器兼容性
 
 {{Compat}}
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:152]: line 156: 表达式与运算符

File: files/zh-cn/web/javascript/reference/statements/index.md [hunk 13]
```diff
@@ -44,48 +44,113 @@ JavaScript 应用程序是由许多语法正确的语句组成的。单个语句
...
-  - : 在异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/JavaScript/Guide/Iterators_and_generators)上迭代，调用自定义迭代钩子，其中包含要为每个不同属性的值执行的语句。
+  - : 遍历异步可迭代对象、类数组对象、[迭代器和生成器](/zh-CN/docs/Web/JavaScript/Guide/Iterators_and_generators)，对每个不同的属性的值，调用一个自定义的带有要执行的语句的迭代钩子。
 - {{jsxref("Statements/while", "while")}}
-  - : 创建一个循环语句，循环会一直持续到该语句条件表达式的值为 false。先执行条件表达式，然后执行语句。
+  - : 创建一个执行指定语句的循环，直到测试条件的求值为假。先对条件进行求值，再执行语句。
 
 ### 其他
 
+- {{jsxref("Statements/Empty", "空语句", "", 1)}}
+  - : 空语句，用于不在 JavaScript 期望语句语法的地方提供语句。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/index.md:73]: The English content does not use the term "statement syntax", it only emphasizes "syntax"

**Review by yin1999 — APPROVED** [review]
:+1: LGTM

**Pull request merged**